### PR TITLE
Add `custom find_operation` support

### PR DESCRIPTION
**Pull request by nnbu**
Issue: aws-controllers-k8s/community#1827

Description of changes:
This change allow to implement custom find function. The name of the function could be written in generator.yaml as 'find_operation: <name-of-function>' similar to update_operation. And generated code for sdkFind() would call this custom find function. The find function itself could be written in hooks.go.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/config/resource.go [hunk 1]
```diff
@@ -62,7 +62,19 @@ type ResourceConfig struct {
...
+	// there is no describe/find/list apis. However, it is possible to write
+	// `find` function for such resources using another resource's (say,
+	// resource B) `find` function. Then, one could write custom code to find
+	// the relevant resource from the response of resource B's find function.
+	//
+	// Additionally, if there the `find` function for a resource is not
+	// available then reconciler's progression is stalled with the error
+	// `find` operation `notImplemented`. Custom find function allows the as
+	// reconciler's progression.
+	FindOperation *CustomOperationsConfig `json:"find_operation,omitempty"`
```
Comment (A-Hilaly) [inline pkg/config/resource.go:77]: To keep consistent naming across all the packages, let's name this `ReadOperation` or `ReadOneOperation` ?

File: pkg/config/resource.go [hunk 2]
```diff
@@ -312,12 +324,12 @@ type ListOperationConfig struct {
 	MatchFields []string `json:"match_fields"`
 }
 
-// UpdateOperationConfig contains instructions for the code generator to handle
-// Update operations for service APIs that have resources that have
-// difficult-to-standardize update operations.
-type UpdateOperationConfig struct {
```
Comment (A-Hilaly) [inline pkg/config/resource.go:318]: Since `omitUnchangedFields` doesn't apply to read operations, we can create a new struct called `ReadOperationConfig`

File: pkg/generate/code/set_resource.go [hunk 3]
```diff
@@ -962,6 +962,8 @@ func SetResourceIdentifiers(
 		case r.Ops.ReadMany != nil:
 			// If single lookups can only be done using ReadMany
 			op = r.Ops.ReadMany
+		default:
+			return ""
```
Comment (A-Hilaly) [inline pkg/generate/code/set_resource.go:966]: Not sure why is this needed, it could also break other functionalities

File: pkg/config/resource.go (hunk 1 above)
Comment (nnbu) [reply pkg/config/resource.go]: done

File: pkg/config/resource.go (hunk 2 above)
Comment (nnbu) [reply pkg/config/resource.go]: done

File: pkg/generate/code/set_resource.go (hunk 3 above)
Comment (nnbu) [reply pkg/generate/code/set_resource.go:966]: `If op == nil` on line 957 is true, then we get into this `switch` statement. However, if both cases `r.Ops.GetAttributes` and `r.Ops.ReadMany` are `nil`, then `op` remains null. and then the code generator crashes on line 969 when it dereferences `op` as `op.InputRef.Shape`.

This crash happened when I was testing `networkAclEntry` which does not have `read` operation and `op` remained `nil`. 

So fixing this already existing bug.

**Review by a-hilaly — APPROVED** [review]
Thank you @nnbu !
/lgtm
/test all
